pom.xml

```diff
@@ -88,6 +88,11 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
+            <dependency>
+                <groupId>com.lmax</groupId>
+                <artifactId>disruptor</artifactId>
+                <version>3.4.4</version>
+            </dependency>
```

Comment:
Is this a complete framework or just a logging component?
What is the link to its repository?

Reply:
[disruptor](https://lmax-exchange.github.io/disruptor/) is a "High Performance Inter-Thread Messaging Library" . Log4j2 uses this lib to implement async logging.

Reply:
With this lib added, will the java runner logs be output in APISIX?

Reply:
It is `log4j2.xml`  decides whether runner's logs be output in  APISIX. Now runners log will not be send to APISIX. If users want to send logs to apisix, they need add following code in `log4j2.xml`:


Add a appender:
```xml
<Console name="CONSOLE" target="SYSTEM_OUT" follow="true">
    <PatternLayout charset="${FILE_ENCODING}">
        <pattern>%d %-5p %-32t - %m%n %throwable</pattern>
    </PatternLayout>
</Console>
```

Add appender above to loggers:
```xml
<appender-ref ref="CONSOLE"/>
```

Reply:
do we need to set the system property `log4j2.contextSelector` to `org.apache.logging.log4j.core.async.AsyncLoggerContextSelector` ref: https://logging.apache.org/log4j/2.x/manual/async.html

Reply:
Already set `<asyncRoot>`, so there is no need to set system property "log4j2.contextSelector" to any value.